_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp

```diff
@@ -1391,14 +1391,37 @@ bool MfxHwH264Encode::IsCmNeededForSCD(
 }
 
 bool MfxHwH264Encode::IsMctfSupported(
-    MfxVideoParam const & video)
+    MfxVideoParam const & video,
+    eMFXHWType            platform)
 {
     bool
         isSupported = false;
 #if defined(MXF_ENABLE_MCTF_IN_AVC)
+    mfxU16
+        max_height = 0;
+    switch (platform)
+    {
+    case MFX_HW_ICL:
+    case MFX_HW_ICL_LP:
+    case MFX_HW_TGL_LP:
+    case MFX_HW_DG1:
+        max_height = 2160;
```

Comment:
We again hardcode some limitations... Is there any way to query these limitations from underlying driver or CM? At least: can we point to the CM/driver source code where it works with these limitations? For example, 1696 is absolutely non-trivial number which I simply don't understand where it came from...

Reply:
I did the test until I found the actual limit, in this case if you use 1698 (even 1697 which is odd but anyway), you get the device_failure. now 2160 is the limit for MCTF and not for the driver.

Reply:
Well, this is weird. So, effectively we were guessing. But what if BXT has different Limit? or ICL? I guess you did not try all the platforms. So, we can accept this change as a workaround, but we need to find out the actual limit driven by implementation (driver or cm or whatever). And optimally this limit should be reported thru caps.

Reply:
Hi @spossos ,

> you get the device_failure

Is this device_failure failure on memory allocation and from kernel execution or ?

Reply:
Talked with @spossos  offline. The talk is about this call:
https://github.com/Intel-Media-SDK/MediaSDK/blob/e7b1363357e22b778e8d0706fdc842a45c2f8580/_studio/mfx_lib/mctf_package/mctf/src/mctf_common.cpp#L3161
It seems we don't understand its limitations. And @spossos guessed them.

I wonder whether we can use msdk internal copy here which hopefully does not have these limitations:
https://github.com/Intel-Media-SDK/MediaSDK/blob/e7b1363357e22b778e8d0706fdc842a45c2f8580/_studio/shared/src/cm_mem_copy.cpp#L1447

Reply:
That's wrong assumption that this gpu copy depends on resolution. See https://github.com/Intel-Media-SDK/MediaSDK/issues/2412#issuecomment-716719434: we clearly fail on the comparison of input and output resolution, output resolution is lower by height. Hence the failure. MCTF simply does not account for possible surface padding alignment.